From c4da692895d01cf281c22efc2c30a6d4fdfb3d21 Mon Sep 17 00:00:00 2001
From: Jinyang He <hejinyang@loongson.cn>
Date: Fri, 11 Aug 2023 16:10:40 +0800
Subject: [PATCH 009/123] LoongArch: Enable gas sort relocs

The md_pre_output_hook creating fixup is asynchronous, causing relocs
may be out of order in .eh_frame. Define GAS_SORT_RELOCS so that reorder
relocs when write_relocs.

Reported-by: Rui Ueyama <rui314@gmail.com>
---
 gas/config/tc-loongarch.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gas/config/tc-loongarch.h b/gas/config/tc-loongarch.h
index d353f18d..fd094356 100644
--- a/gas/config/tc-loongarch.h
+++ b/gas/config/tc-loongarch.h
@@ -123,6 +123,7 @@ extern void tc_loongarch_parse_to_dw2regnum (expressionS *);
 
 extern void loongarch_pre_output_hook (void);
 #define md_pre_output_hook loongarch_pre_output_hook ()
+#define GAS_SORT_RELOCS 1
 
 #define SUB_SEGMENT_ALIGN(SEG, FRCHAIN) 0
 
-- 
2.33.0

